components/ai_chat/core/common/test_mojom_printers.cc

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/ai_chat/core/common/test_mojom_printers.h"
+
+#include <utility>
+
+#include "base/base64.h"
+#include "base/i18n/time_formatting.h"
+#include "base/json/json_writer.h"
+#include "base/json/values_util.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
+
+namespace ai_chat {
+namespace mojom {
+
+void PrintTo(const AssociatedContent& content, std::ostream* os) {
+  *os << "--AssociatedContent--\n";
+  *os << "  uuid: " << content.uuid << "\n";
+  *os << "  title: " << content.title << "\n";
+  *os << "  content_id: " << content.content_id << "\n";
+  *os << "  url: " << content.url.possibly_invalid_spec() << "\n";
+  *os << "  content_used_percentage: " << content.content_used_percentage
+      << "\n";
+  *os << "  content_type: " << static_cast<int>(content.content_type) << "\n";
```

Comment:
can we use `base::StringPrintf` with a template so it would be easier to read?
ex.
```
 *os << base::StringPrintf(
      R"(--AssociatedContent--
  uuid: %s
  title: %s
  content_id: %d
  url: %s
  content_used_percentage: %d
  content_type: %d
)",
      content.uuid.c_str(),
      content.title.c_str(),
      content.content_id,
      content.url.possibly_invalid_spec().c_str(),
      content.content_used_percentage,
      static_cast<int>(content.content_type));
 ```

Reply:
Is this a nit? Because I personally find the current way a bit easier to read and also write, especially for many params in order to avoid counting when modifying or adding new fields, as well as avoiding mistakes. As well as easier to have the conditional parts. This is a best attempt to make the formatting nice, but the important thing is to make sure as many fields are represented to that when equality comparison fails, the developer has a chance of quickly knowing what's different.